src/Microsoft.IdentityModel.Tokens/SignatureProvider.cs

```diff
@@ -41,7 +41,7 @@ public abstract class SignatureProvider : IDisposable
         /// <param name="key">The <see cref="SecurityKey"/> that will be used for signature operations.</param>
         /// <param name="algorithm">The signature algorithm to apply.</param>
         /// <exception cref="ArgumentNullException">'key' is null.</exception>
-        protected SignatureProvider(SecurityKey key, string algorithm)
+        public SignatureProvider(SecurityKey key, string algorithm)
```

Comment:
This should not be public. We can actually get rid of this ctor. All it does it null key check and couple of assignments. We, anyways, do the check in derived classes, so this is not really useful.